File: Leetcode/976.py
"""
976. Largest Perimeter Triangle
Example 1:

Input: nums = [2,1,2]
Output: 5
Example 2:

Input: nums = [1,2,1]
Output: 0


Constraints:

3 <= nums.length <= 104
1 <= nums[i] <= 106
"""
from typing import List


def largestPerimeter(nums: List[int]) -> int:
    nums.sort()
    res = 0
    for i in range(len(nums) - 2):
        if nums[i] + nums[i + 1] > nums[i + 2]:
            res = max(res, nums[i] + nums[i + 1] + nums[i + 2])
    return res

File: Leetcode/976.go
package main

import "sort"

func largestPerimeter(nums []int) int {
	res := 0
	sort.Ints(nums)
	for i := 1; i < len(nums)-1; i++ {
		if nums[i-1]+nums[i] > nums[i+1] {
			res = maxInt(res, nums[i-1]+nums[i]+nums[i+1])
		}
	}
	return res
}
func maxInt(a int, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}
